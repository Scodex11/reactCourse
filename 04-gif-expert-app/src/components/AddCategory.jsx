import { useState } from 'react'
import PropTypes from 'prop-types'

/* DESCRIPCIÓN del COMPONTENTE */
// Componente que renderiza un formulario para agregar una nueva categoría y se llama desde GiftExpertApp

export const AddCategory = ({ onNewCategory }) => {
  
    const [inputValue, setInputValue] = useState('')
  
    const onInputChange = ({target}) => {
        setInputValue(target.value); //Enviamos el valor obtenido por teclado al useState
    }

    
    const onSubmit = (event) => {
        event.preventDefault(); //Evita que refresque 
        
        if (inputValue.trim().length <= 1) return;

        setInputValue('')
        onNewCategory(inputValue.trim()); //Enviamos el valor del useState al componente padre (GifExpertApp)
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input  
                type="text"
                placeholder="Escribe la categoria del gif..."
                value={inputValue} //Valor del useState
                onChange={ onInputChange } //Método que permite cambiar el estado de value dentro de el input
            />
        </form>
  
  )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
