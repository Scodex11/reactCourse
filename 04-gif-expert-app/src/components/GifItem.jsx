import PropTypes from 'prop-types';

/* DESCRIPCIÃ“N */
// Componente que renderiza una CARD y es utilizada en el componente GifGrid

export const GifItem = ({title, url}) => {
  return (
    
    <div className="card">
      <img src={url} alt={ title }/>
      <p>{title}</p>
    </div>

  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}


