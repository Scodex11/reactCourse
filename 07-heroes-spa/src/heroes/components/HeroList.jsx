import { useMemo } from "react";
import { getHeroesByPublisher } from "../helpers";
import { HeroCard } from "./";


export const HeroList = ({ publisher }) => {
  const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher])

  
  return (
    <>
      <ul>
        {
            heroes.map((heroe)=> (
                <HeroCard key={heroe.id} {...heroe} />
            ))
        }
      </ul>
    </>
  );
};
