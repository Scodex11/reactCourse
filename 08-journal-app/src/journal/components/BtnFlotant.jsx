import { Add } from "@mui/icons-material";
import { Fab } from "@mui/material";

export const BtnFlotant = () => {
    return (
        <Fab color="primary" aria-label="add" style={{ position: 'fixed', bottom: 16, right: 16 }}>
          <Add />
        </Fab>
      );
}
