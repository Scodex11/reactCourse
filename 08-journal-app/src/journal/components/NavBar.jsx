import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material";
import { LogoutOutlined, MenuOutlined } from "@mui/icons-material";

export const NavBar = ({ drawerWidth }) => {
  return (
    <AppBar
      position="fixed" //va a estar fijo siempre
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` }, //si es sm va a ser el 100% menos el drawerWidth
        ml: { sm: `${drawerWidth}px` }, //si es sm va a ser el drawerWidth
      }}
    >
      {/* Toolbar */}
      <Toolbar>
        <IconButton
          color="inherit"
          edge="start"
          sx={{ mr: 2, display: { sm: "none" } }}
        >
          <MenuOutlined />
        </IconButton>

        <Grid
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Typography variant="h6" noWrap component='div' >Journal App</Typography>

          <IconButton color="error">
            <LogoutOutlined />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
