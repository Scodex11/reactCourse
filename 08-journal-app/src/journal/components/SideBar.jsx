import { Box,AppBar, Drawer, Grid, IconButton, Toolbar, Typography, Divider, List, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material";
import { LogoutOutlined, MenuOutlined, TurnedInNot } from "@mui/icons-material";

const longText = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const SideBar = ({ drawerWidth }) => {
  return (
    <Box 
      component = 'nav'
      sx = {{
        width: { sm: `${drawerWidth}px` },
        flexShrink: { sm: 0 },
      }}
    >
        <Drawer
          variant="permanent"
          open
          sx={{
            display: { xs: "block" },
            "& .MuiDrawer-paper": {
              boxSizing: "border-box",
              width: drawerWidth,
            },
          }}
        >

          <Toolbar>
            <Typography variant="h6" noWrap component='div' >Facundo Rodriguez</Typography>
          </Toolbar>
          <Divider />

          <List>
            {
              [1,2,3,4,5,6,7,8,9,10].map( text => (
                <ListItem key={text} disablePadding>
                  <ListItemButton>
                    <ListItemIcon>
                      <TurnedInNot />
                    </ListItemIcon>

                    <Grid container >
                      <ListItemText primary = {text} />
                      <ListItemText secondary ={longText.slice(0, 30) + '...'}/>

                    </Grid>

                  </ListItemButton>
                </ListItem>
              ))
            }
          </List>


        </Drawer>
    </Box>
  );
};
