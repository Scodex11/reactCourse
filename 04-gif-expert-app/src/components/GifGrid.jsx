import PropTypes from 'prop-types';
import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

//DESCRIPCION del COMPONENTE:
//Este componente se encarga de colocar en un GRID los GIF item que se obtienen de la API

export const GifGrid = ({category}) => {

    const { images, isLoading } = useFetchGifs(category);


    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && <p>Cargando...</p>
            }
            
            <div className="card-grid">
                {
                    images.map( (img) => (
                        <GifItem 
                            key={img.id}
                            {...img} // Enviamos todas las propiedades al componente
                        />
                    ))
                }
            </div>
            
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}