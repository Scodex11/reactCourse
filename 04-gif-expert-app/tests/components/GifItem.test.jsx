import { render, screen} from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';


describe('Pruebas en <GifItem />', () => { 
    
    const title = 'Japanese';
    const url = 'https://localhost/algo.jpg';

    test('debe de mostrar el componente correctamente', () => {
      
        const {container} = render(<GifItem title={ title } url={ url }/>)
        
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render(<GifItem title={ title } url={ url }/>)
        
        //Desestructuramos el primer valor con img  obtener SRC y ALT
        const {src, alt} = screen.getByRole('img');
        
        expect(src).toBe(url);
        expect(alt).toBe(title);
     })

     test('El titulo debe existir', () => { 
        
        render(<GifItem title={ title } url={ url }/>)
        
        const titleElement = screen.getByText(title);
        
        expect(titleElement).toBeTruthy();

      })


 })