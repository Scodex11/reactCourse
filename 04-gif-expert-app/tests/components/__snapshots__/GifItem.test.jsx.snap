// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Japanese"
      src="https://localhost/algo.jpg"
    />
    <p>
      Japanese
    </p>
  </div>
</div>
`;
